### PR TITLE
new command: qrencode-url

### PR DESCRIPTION
**Pull request by Hugo-Heagren**
This is really cool package and I wanted to use it in my embark setup -- for displaying arbitrary URLs as QR codes. The easisest way to do this is with a simple interactive command which just takes a URL and displays it. I thought others might find it useful too.

### CODE REVIEW

File: qrencode.el [hunk 1]
```diff
@@ -1084,6 +1084,15 @@ Commands:
         (message "No URL found at point")
       (qrencode--encode-to-buffer url))))
 
+;;;###autoload
+(defun qrencode-url (url)
+  "Encode URL.
+When called interacively, read URL with `read-string'."
+  (interactive `(,(read-string "URL: ")))
```
Comment (ruediger) [inline qrencode.el:1091]: Why the macro use and read-string? Wouldn't `(interactive "sURL: ")` achieve the same?

File: qrencode.el [hunk 2]
```diff
@@ -1084,6 +1084,15 @@ Commands:
...
+  (if (null (string-trim url))
```
Comment (ruediger) [inline qrencode.el:1092]: `(null "")` returns `nil`. Maybe you mean `string-empty-p`?

File: qrencode.el [hunk 3]
```diff
@@ -1084,6 +1084,15 @@ Commands:
...
+      (user-error "No URL entered")
+    (qrencode--encode-to-buffer url)))
```
Comment (ruediger) [inline qrencode.el:1094]: It would probably be good to pass the string-trim version here as well, as this would otherwise encode the spaces as well.

**Review by ruediger — COMMENTED** [review]
Thank you for your contribution.

File: qrencode.el (hunk 1 above)
Comment (Hugo-Heagren) [reply qrencode.el]: Thanks -- I've been writing a lot of stuff recently with weird interactive inputs -- forgot about the string codes. Fixed.

File: qrencode.el (hunk 3 above)
Comment (Hugo-Heagren) [reply qrencode.el]: Done.

File: qrencode.el (hunk 2 above)
Comment (Hugo-Heagren) [reply qrencode.el]: Done.

**Comment by Hugo-Heagren** [comment]
I've renamed the command `qrencode-string`, because it doesn't require that the string is a URL.

File: qrencode.el [hunk 4]
```diff
@@ -1084,6 +1084,16 @@ Commands:
...
       (qrencode--encode-to-buffer url))))
 
+;;;###autoload
+(defun qrencode-string (str)
+  "Encode STR.
+When called interactively, read STR with `read-string'."
+  (interactive "sString: ")
+  (let ((trimmed-str (string-trim str)))
+    (if (string-empty-p trimmed-str)
+	(user-error "No URL entered"))
```
Comment (ruediger) [inline qrencode.el:1094]: Nit: the user-error should probably say "Empty string" now that the function is renamed.

**Comment by ruediger** [comment]
One minor nit. Then this is ready to get merged. Thank you!

File: qrencode.el (hunk 4 above)
Comment (Hugo-Heagren) [reply qrencode.el:1094]: Good catch.

Comment (Hugo-Heagren) [reply qrencode.el]: Just fixed -- thanks!

**Comment by ruediger** [comment]
Thank you! Sorry for the late response.

**Pull request merged**